src/Sulu/Bundle/MediaBundle/Media/FormatCache/LocalFormatCache.php

```diff
@@ -198,13 +198,11 @@ protected function getFormatFromUrl($url)
     /**
      * Return the requested image format from url.
      *
-     * @param string $url
-     *
-     * @return int
+     * @return string
      *
      * @throws ImageProxyInvalidUrl
      */
-    protected function getFileNameFromUrl($url)
+    protected function getFileNameFromUrl(string $url)
```

Comment:
Adding typehints is for bc reasons not allowed for protected or public methods:

```suggestion
    protected function getFileNameFromUrl($url)
```

Reply:
Please correct my if I am wrong, but as far as I know, contravariant parameter types are safe. If the overriding class did not specify a parameter type there is no issue (see [example](https://3v4l.org/4Lj2s)), because then its type is wider than the string type.

Adding a return type is a BC problem of course.

Reply:
Oh that should work. Remember there were some bugs in some PHP 7.x versions but as we are on 8.0 for Sulu 2.5 nobody should get into that bug longer.